File: tmdb-core/src/main/java/dev/reuss/tmdb/core/exception/TmdbMappingException.java
package dev.reuss.tmdb.core.exception;

/**
 * Exception thrown when a TMDB response cannot be mapped to the expected model.
 */
public class TmdbMappingException extends TmdbException {

    public TmdbMappingException(String message) {
        super(message);
    }

    public TmdbMappingException(String message, Throwable cause) {
        super(message, cause);
    }
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/core/exception/TmdbMappingException.kt
package dev.reuss.tmdb.core.exception

/**
 * Exception thrown when a TMDB response cannot be mapped to the expected model.
 *
 * @property message detail message
 * @property cause cause of the failure
 */
class TmdbMappingException @JvmOverloads constructor(
    message: String,
    cause: Throwable? = null
) : TmdbException(message, cause)